### PR TITLE
Concat and log and stuff

### PR DESCRIPTION
**Pull request by renato-scaroni**

### CODE REVIEW

File: Implementations/Algorithms/Pong_REINFORCE.py [hunk 1]
```diff
@@ -53,41 +55,57 @@ def forward(self, state):
...
         if self.torch_rand:
-            logits = self.policy_network(state)
-            dist = Categorical(logits=logits)
+            probs = self.policy_network(state)
+            dist = Categorical(probs=probs)
             highest_prob_action = dist.sample()
             log_prob = dist.log_prob(highest_prob_action)
         else:
-            probs = self.policy_network(state).cpu()
+            probs = self.policy_network(state)
```
Comment (RenatoGeh) [inline Implementations/Algorithms/Pong_REINFORCE.py]: If you run this on the GPU, won't it spit out an error and complain about needing to access it from the host memory?

File: Implementations/Algorithms/Pong_REINFORCE.py [hunk 2]
```diff
@@ -140,7 +158,7 @@ def update_policy(self, rewards, log_probs, last_g, last_b, last_log_probs):
         discounted_rewards = (discounted_rewards - discounted_rewards.mean()) / (discounted_rewards.std() + 1e-9) # normalize discounted rewards
 
         policy_gradient = []
-        for log_prob, Gt in zip_longest(log_probs, discounted_rewards):
+        for log_prob, Gt in zip(log_probs, discounted_rewards):
```
Comment (RenatoGeh) [inline Implementations/Algorithms/Pong_REINFORCE.py:161]: When not using reward-to-go, the above change ends up truncating to the shortest of the two lists. That's not a desired outcome for this case, as we want `policy_gradient` to compute the sum of all `log_probs`.

File: Implementations/Algorithms/Pong_REINFORCE.py (hunk 1 above)
Comment (renato-scaroni) [reply Implementations/Algorithms/Pong_REINFORCE.py]: Done!

File: Implementations/Algorithms/Pong_REINFORCE.py (hunk 2 above)
Comment (renato-scaroni) [reply Implementations/Algorithms/Pong_REINFORCE.py:161]: done
